### PR TITLE
Add from_csv support to automatically extract columns and dtypes

### PR DESCRIPTION
**Pull request by yongtang**
This PR adds the from_csv support to automatically extract column names
and dtypes without user specifying it.

The function only works in eager mode though.

Signed-off-by: Yong Tang <yong.tang.github@outlook.com>

### CODE REVIEW

File: tensorflow_io/text/python/ops/text_ops.py [hunk 1]
```diff
@@ -77,3 +81,53 @@ def __init__(self, filenames):
...
+  Args:
+    filename: A `tf.string` tensor containing filename.
+  """
+  if not tf.executing_eagerly():
+    raise NotImplementedError("from_csv only support eager mode")
+  dataset = TextDataset(filename)
+  columns = None
+  if header is not None:
+    if header != 0:
+      raise NotImplementedError("from_csv only header=0 or header=None for now")
```
Comment (terrytangyuan) [inline tensorflow_io/text/python/ops/text_ops.py]: only -> only supports

File: tensorflow_io/text/python/ops/text_ops.py [hunk 2]
```diff
@@ -77,3 +81,53 @@ def __init__(self, filenames):
...
+
+  Args:
+    filename: A `tf.string` tensor containing filename.
+  """
+  if not tf.executing_eagerly():
+    raise NotImplementedError("from_csv only support eager mode")
+  dataset = TextDataset(filename)
+  columns = None
+  if header is not None:
+    if header != 0:
```
Comment (terrytangyuan) [inline tensorflow_io/text/python/ops/text_ops.py]: combine these two if conditions?

File: tensorflow_io/text/python/ops/text_ops.py [hunk 3]
```diff
@@ -77,3 +81,53 @@ def __init__(self, filenames):
...
+def from_csv(filename, header=0):
+  """Read csv to Dataset
+
+  NOTE: Experimental and eager only!
+
+  Args:
+    filename: A `tf.string` tensor containing filename.
+  """
+  if not tf.executing_eagerly():
+    raise NotImplementedError("from_csv only support eager mode")
```
Comment (terrytangyuan) [inline tensorflow_io/text/python/ops/text_ops.py]: support -> supports

File: tests/test_text_eager.py [hunk 4]
```diff
@@ -148,5 +148,43 @@ def test_csv_output():
     i += 1
   assert i == 5
 
+def test_from_csv():
+  """test_from_csv"""
```
Comment (terrytangyuan) [inline tests/test_text_eager.py]: Better to have a complete sentence

File: tensorflow_io/text/python/ops/text_ops.py (hunk 1 above)
Comment (yongtang) [reply tensorflow_io/text/python/ops/text_ops.py]: @terrytangyuan Thanks. Done.

File: tensorflow_io/text/python/ops/text_ops.py (hunk 2 above)
Comment (yongtang) [reply tensorflow_io/text/python/ops/text_ops.py]: @terrytangyuan The two conditions are not exactly combined. As:
- header=None: no column specified, so using `range(number of columns)` as column names
- header=0: Use the first line as header
- header=1+: tf.data.Dataset does not exactly help in this case, so a ValueError has been raised.

File: tensorflow_io/text/python/ops/text_ops.py (hunk 3 above)
Comment (yongtang) [reply tensorflow_io/text/python/ops/text_ops.py]: @terrytangyuan  Done. Thanks.

File: tensorflow_io/text/python/ops/text_ops.py (hunk 2 above)
Comment (yongtang) [reply tensorflow_io/text/python/ops/text_ops.py]: @terrytangyuan this follows pandas's `read_csv`'s convention.

File: tests/test_text_eager.py (hunk 4 above)
Comment (yongtang) [reply tests/test_text_eager.py]: @terrytangyuan Done.

**Comment by yongtang** [comment]
@terrytangyuan The PR has been updated. Please take a look.

**Pull request merged**